### PR TITLE
fix(core): Unescape variable strings before dictionary lookup in `EncodedVariableInterpreter::encode_and_search_dictionary` (fixes #590).

### LINKED ISSUE #590: clp-s: Several issues searching for logs that contain escaped characters.
### Bug

Search does not work as expected when searching against JSON values that contain escaped characters. This is likely an issue with how string predicates are un-escaped both for clp style search, and for wildcard matching.

Importantly, clp-s makes the decision to not un-escape raw JSON values before ingesting them, which causes some edge cases we are not currently considering during search.

For example, the value in `{"key": "a: \"bcde\""}` gets ingested verbatim as `a: \"bcde\"`. However, the search `*: "a: \"bcde\"` fails to return the matching result.

### CLP version

0.1.2

### Environment

clp-text package.

### Reproduction steps

Ingest `{"key": "a: \"bcde\""}`
Perform the query `*: "a: \"bcde\"`

### PR DESCRIPTION
**Pull request by gibber9809**
# Description

This PR fixes a bug in `EncodedVariableInterpreter::encode_and_search_dictionary` where variable strings were not unescaped before being looked up in the variable dictionary. Fixing this issue addresses the last outstanding failing case in #590 (note however that we should also finish updating our wildcard search implementation to solve similar unescaping bugs).

The bug this PR solves could be encountered when issuing searches that target variable dictionary entries containing the `\` character -- since `\` isn't a delimiter (unlike other characters that need to be escaped for literal search such as `*`, `?`, and the other variable placeholders) the heuristic parser will parse a string like `"text \a123"` as the logtext `"text \var"` and the variable `"\a123"`. However, to query such a string the user needs to escape the `\`, i.e. the query string will be `"text \\a123"`; to properly perform the dictionary lookup the escaped variable `"\\a123"` must be unescaped as `"\a123"`.

This change is implemented by adding a simple unescaping utility to string utils which we use in `encode_and_search_dictionary`, as well as one other place in clp-s that was previously unescaping query strings in line before doing dictionary lookups.

We expand testing in `test-EncodedVariableInterpreter.cpp` to cover this edge case, and enable the failing case from #590 in `test-clp_s-search.cpp`.

# Checklist

* [x] The PR satisfies the [contribution guidelines][yscope-contrib-guidelines].
* [x] This is a breaking change and that has been indicated in the PR title, OR this isn't a
  breaking change.
* [x] Necessary docs have been updated, OR no docs need to be updated.

# Validation performed
* Expanded test in `test-EncodedVariableInterpreter.cpp`
* Enabled test in `test-clp_s-search.cpp`

[yscope-contrib-guidelines]: https://docs.yscope.com/dev-guide/contrib-guides-overview.html

## Summary by CodeRabbit

* **Bug Fixes**
  * Searches and dictionary lookups now unescape input so queries match values containing escaped characters (e.g., backslashes); wildcard results now include entries like “\Abc123”.

* **New Features**
  * Centralized unescape handling for query strings to ensure consistent matching of escaped content across search operations.

* **Tests**
  * Re-enabled a previously skipped search case and added extensive tests covering escaped-character scenarios and unescape behaviour.

### CODE REVIEW

File: components/core/tests/test-EncodedVariableInterpreter.cpp [hunk 1]
```diff
@@ -530,6 +536,30 @@ TEST_CASE("EncodedVariableInterpreter", "[EncodedVariableInterpreter]") {
...
+                    || enum_to_underlying_type(VariablePlaceholder::Integer) == c
+                    || enum_to_underlying_type(VariablePlaceholder::Float) == c
+                    || enum_to_underlying_type(VariablePlaceholder::Dictionary) == c)
+                {
+                    escaped_string.push_back(enum_to_underlying_type(VariablePlaceholder::Escape));
+                }
+                escaped_string.push_back(c);
+            }
+            return escaped_string;
+        };
```
[the remark this answers is not in the document]
Comment (gibber9809) [reply components/core/tests/test-EncodedVariableInterpreter.cpp:553]: Not sufficient since query escaping has different rules than logtype escaping.

File: components/core/src/clp/string_utils/string_utils.hpp [hunk 2]
```diff
@@ -87,6 +88,17 @@ void to_lower(std::string& str);
...
+/**
+ * Unescapes a string according to the following rules:
+ * <ul>
+ *   <li>Escape sequences `\<char>` are replaced by `<char>`</li>
+ *   <li>Lone dangling `\` is removed from the end of the string</li>
+ * </ul>
+ * @param str
+ * @return An unescaped version of `str`.
+ */
+auto unescape_string(std::string_view str) -> std::string;
```
Comment (LinZhihao-723) [inline components/core/src/clp/string_utils/string_utils.hpp]:
```suggestion
[[nodiscard]] auto unescape_string(std::string_view str) -> std::string;
```
